Data/HashMap/Internal.hs

```diff
@@ -1751,6 +1752,18 @@ traverseWithKey f = go
         Collision h <$> A.traverse' (\ (L k v) -> L k <$> f k v) ary
 {-# INLINE traverseWithKey #-}
 
+-- | /O(n)/.
+-- @'mapKeys' f s@ is the map obtained by applying @f@ to each key of @s@.
+--
+-- The size of the result may be smaller if @f@ maps two or more distinct
+-- keys to the same new key.  In this case the later mappings take precedence.
```

Comment:
Unfortunately there's no guarantee about the order of the mappings. The `u-c` internals or `Hashable` instances can change this at any time.

So the results would only be predictable with an injective function. IMHO that might be too much of a pitfall. How about only adding `mapKeysWith`? Even then the results would only be deterministic when the combining function is both commutative and associative… :/

So what's your use case?

Reply:
I feared there could be such complications.

My use case is simply wrapping the keys into newtypes, which uses injective functions.

Adding only `mapKeysWith` does not look to me a nicer idea, since it still leave space for unexpected behaviours.

Either we add `mapKeys` and we document the fact the with non-injective functions the behaviour could be unexpected, or we just close this PR and we leave `mapKeys` out of `u-c`

Reply:
Since this library already contains so many functions where the result depends on the unspecified order of the key-value pairs, I don't see sufficient grounds for rejecting this one. Can you add a bit of documentation?